mlrun/projects/project.py

```diff
@@ -1907,15 +1921,26 @@ def run(
         if workflow_path or (workflow_handler and callable(workflow_handler)):
             workflow_spec = WorkflowSpec(path=workflow_path, args=arguments)
         else:
-            workflow_spec = WorkflowSpec.from_dict(self.spec._workflows[name])
+            workflow_spec = self.spec._workflows[name].copy()
```

Comment:
Keep in mind that `copy` do not copy lists and dicts. If the spec has those kind of field its better to use deepcopy.